pkg/leader/leader.go

```diff
@@ -109,7 +109,33 @@ func Become(ctx context.Context, lockName string) error {
 			log.Info("Became the leader.")
 			return nil
 		case apierrors.IsAlreadyExists(err):
-			log.Info("Not the leader. Waiting.")
+			existing := &corev1.ConfigMap{}
+			key := crclient.ObjectKey{Namespace: ns, Name: lockName}
+			err := client.Get(ctx, key, existing)
+			if err != nil {
+				return err
+			}
+			leaderPod := &corev1.Pod{}
+			key = crclient.ObjectKey{Namespace: ns, Name: existing.ObjectMeta.OwnerReferences[0].Name}
+			err = client.Get(ctx, key, leaderPod)
+			if err != nil {
+				return err
+			}
+
+			podFailed := leaderPod.Status.Phase == corev1.PodFailed
```

Comment:
wdyt about export a func IsPodFailed and isPodEvicted?

Reply:
@camilamacedo86 What would be the motivation for exporting these new functions if this is the only place we only need these values?

Also, keep in mind that we are trying to move all exported Operator SDK operator functionality to controller-runtime, so we should be trying to decrease the API surface of Operator SDK, not increasing it.

Reply:
Hi @joelanford, 

My suggestion is just to extract the complexity and make local func

```
if isPodFailed(.. ) && isPodEvicted() {

...

fun  isPodFailed() bool{
   return ...
}
```

Reply:
Ah okay. In Go, to export an identifier means to make it public by capitalizing its first letter. If you're just suggesting that we make these private unexported functions, then I'd be in agreement as a nit.

Reply:
I saw that one of the `is` was written in Capital letter. Sorry, for the misunderstanding, the idea was not suggesting to allow it to be used in other places just a nit to make more clear the code.